common/lib/plugin_manager.ts

```diff
@@ -282,20 +283,26 @@ export class PluginManager {
   }
 
   getHostInfoByStrategy(role: HostRole, strategy: string, hosts?: HostInfo[]): HostInfo {
-    for (const plugin of this._plugins) {
-      const pluginSubscribedMethods = plugin.getSubscribedMethods();
-      const isSubscribed =
-        pluginSubscribedMethods.has(PluginManager.ALL_METHODS) || pluginSubscribedMethods.has(PluginManager.GET_HOST_INFO_BY_STRATEGY_METHOD);
-
-      if (isSubscribed) {
-        try {
-          const host = plugin.getHostInfoByStrategy(role, strategy, hosts);
-          if (host) {
-            return host;
-          }
-        } catch (error) {
-          // This plugin does not support the provided strategy, ignore the exception and move on
+    if (PluginManager.GET_HOST_INFO_BY_STRATEGY_PLUGIN_CHAIN_CACHE.length === 0) {
```

Comment:
Could you please use `PLUGIN_CHAIN_CACHE`'s implementation as a reference?
Different connections may be using different list of plugins, your current implementation will throw an error.
See this simple test:
```ts
    const client = new AwsPGClient({
      user: PG_DB_USER,
      host: PG_DB_HOST,
      database: PG_DB_NAME,
      password: PG_DB_PASSWORD,
      port: 5432,
      plugins: "readWriteSplitting",
      readerHostSelectorStrategy: "leastConnections",
      connectionProvider: new InternalPooledConnectionProvider()
    });

    await client.connect();

    const client2 = new AwsPGClient({
      user: PG_DB_USER,
      host: PG_DB_HOST,
      database: PG_DB_NAME,
      password: PG_DB_PASSWORD,
      port: 5432,
      plugins: ""
    });

    await client2.connect();
```

Reply:
Also this implementation will iterate over the plugin chain twice on the initial connection. This can be optimized.